tools/topology/sof/pipe-dcblock-volume-playback.m4

```diff
@@ -0,0 +1,99 @@
+# Low Latency Passthrough with volume Pipeline and PCM
+#
+# Pipeline Endpoints for connection are :-
+#
+#  host PCM_P --> B0 --> Volume 0 --> B1 --> sink DAI0
+
+# Include topology builder
+include(`utils.m4')
+include(`buffer.m4')
+include(`pcm.m4')
+include(`pga.m4')
+include(`dai.m4')
+include(`mixercontrol.m4')
+include(`pipeline.m4')
+
+#
+# Controls
+#
+# Volume Mixer control with max value of 32
+C_CONTROLMIXER(Master Playback Volume, PIPELINE_ID,
+	CONTROLMIXER_OPS(volsw, 256 binds the mixer control to volume get/put handlers, 256, 256),
+	CONTROLMIXER_MAX(, 32),
+	false,
+	CONTROLMIXER_TLV(TLV 32 steps from -64dB to 0dB for 2dB, vtlv_m64s2),
+	Channel register and shift for Front Left/Right,
+	LIST(`	', KCONTROL_CHANNEL(FL, 1, 0), KCONTROL_CHANNEL(FR, 1, 1)))
+
+#
+# Volume configuration
+#
+
+W_VENDORTUPLES(playback_pga_tokens, sof_volume_tokens,
+LIST(`		', `SOF_TKN_VOLUME_RAMP_STEP_TYPE	"0"'
+     `		', `SOF_TKN_VOLUME_RAMP_STEP_MS		"250"'))
+
+W_DATA(playback_pga_conf, playback_pga_tokens)
+
+include(`dcblock_coef_default.m4')
+# DC Block Bytes control with max value of 180
+# The max size needs to also take into account the space required to hold the control data IPC message
+# struct sof_ipc_ctrl_data requires 92 bytes
+# DCBLOCK priv in amp_bytes.m4 (ABI header (32 bytes) + 12 dwords) requires 80 bytes
+# Therefore at least 172 bytes are required for this kcontrol
+# Any value lower than that would end up in a topology load error
+ C_CONTROLBYTES(DCBLOCK, PIPELINE_ID,
```

Comment:
Here also use a pipeline specific defined name for "DCBLOCK". Then refer later with that name. If you use "DCBLOCK_priv" below as macro name it's OK.

I noticed that for EQs it was a mistake to try to name the X_priv with some response specific names. I will do a PR later to fix them to the scheme I'm now proposing for DCBLOCK. For IIR e.g. the tuning tuning should create blobs with just name "EQIIR_PRIV". And that is defined to be pipeline specic.

Reply:
@singalsu Updated